File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/reflection/ItemProxy.java
package dev.slne.surf.surfapi.bukkit.server.reflection;

import dev.slne.surf.surfapi.core.api.reflection.annontation.Field;
import dev.slne.surf.surfapi.core.api.reflection.annontation.Field.Type;
import dev.slne.surf.surfapi.core.api.reflection.annontation.SurfProxy;
import net.minecraft.core.component.DataComponentMap;
import net.minecraft.world.item.Item;

@SurfProxy(Item.class)
public interface ItemProxy {

  @Field(name = "components", type = Type.SETTER, overrideFinal = true)
  void setComponents(Item item, DataComponentMap components);
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/reflection/ItemProxy.kt
package dev.slne.surf.surfapi.bukkit.server.reflection

import dev.slne.surf.surfapi.core.api.reflection.Field
import dev.slne.surf.surfapi.core.api.reflection.SurfProxy
import net.minecraft.core.component.DataComponentMap
import net.minecraft.world.item.Item

@SurfProxy(Item::class)
interface ItemProxy {
    @Field("components", Field.Type.SETTER, overrideFinal = true)
    fun setComponents(item: Item, components: DataComponentMap)
}
